executor/importer/import.go

```diff
@@ -258,51 +249,83 @@ func NewPlanFromLoadDataPlan(userSctx sessionctx.Context, plan *plannercore.Load
 		plan.OnDuplicate != ast.OnDuplicateKeyHandlingIgnore
 
-	p := &Plan{
-		DBName:      plan.Table.Schema.O,
-		Path:        plan.Path,
-		Format:      LoadDataFormatDelimitedData,
-		Restrictive: restrictive,
-		IgnoreLines: plan.IgnoreLines,
-		Charset:     charset,
+	var ignoreLines uint64
+	if plan.IgnoreLines != nil {
+		ignoreLines = *plan.IgnoreLines
 	}
-	return p, nil
+
+	var (
+		nullDef              []string
+		nullValueOptEnclosed = false
+	)
+
+	lineFieldsInfo := plannercore.NewLineFieldsInfo(plan.FieldsInfo, plan.LinesInfo)
+	// todo: move null defined into plannercore.LineFieldsInfo
+	// in load data, there maybe multiple null def, but in SELECT ... INTO OUTFILE there's only one
+	if plan.FieldsInfo != nil && plan.FieldsInfo.DefinedNullBy != nil {
+		nullDef = append(nullDef, *plan.FieldsInfo.DefinedNullBy)
+		nullValueOptEnclosed = plan.FieldsInfo.NullValueOptEnclosed
+	} else if len(lineFieldsInfo.FieldsEnclosedBy) != 0 {
+		nullDef = append(nullDef, "NULL")
+	}
+	if len(lineFieldsInfo.FieldsEscapedBy) != 0 {
+		nullDef = append(nullDef, string([]byte{lineFieldsInfo.FieldsEscapedBy[0], 'N'}))
+	}
+
+	return &Plan{
+		DBName: plan.Table.Schema.O,
+		DBID:   plan.Table.DBInfo.ID,
+
+		Path:                 plan.Path,
+		Format:               DataFormatDelimitedData,
+		Restrictive:          restrictive,
+		FieldNullDef:         nullDef,
+		NullValueOptEnclosed: nullValueOptEnclosed,
+		LineFieldsInfo:       lineFieldsInfo,
+		IgnoreLines:          ignoreLines,
+
+		SQLMode:          userSctx.GetSessionVars().SQLMode,
+		Charset:          charset,
+		ImportantSysVars: getImportantSysVars(userSctx),
+
+		DistSQLScanConcurrency: userSctx.GetSessionVars().DistSQLScanConcurrency(),
+	}, nil
 }
 
 // NewImportPlan creates a new import into plan.
 func NewImportPlan(userSctx sessionctx.Context, plan *plannercore.ImportInto, tbl table.Table) (*Plan, error) {
-	fullTableName := common.UniqueTable(plan.Table.Schema.L, plan.Table.Name.L)
-	logger := log.L().With(zap.String("table", fullTableName))
 	var format string
 	if plan.Format != nil {
 		format = strings.ToLower(*plan.Format)
 	} else {
-		// without FORMAT 'xxx' clause, default to DELIMITED DATA
-		format = LoadDataFormatDelimitedData
-	}
-	// todo: use charset in options
-	var charset *string
-	// https://dev.mysql.com/doc/refman/8.0/en/load-data.html#load-data-character-set
-	d, err2 := userSctx.GetSessionVars().GetSessionOrGlobalSystemVar(
-		context.Background(), variable.CharsetDatabase)
-	if err2 != nil {
-		logger.Error("LOAD DATA get charset failed", zap.Error(err2))
-	} else {
-		charset = &d
+		// without FORMAT 'xxx' clause, default to CSV
+		format = DataFormatCSV
 	}
 	restrictive := userSctx.GetSessionVars().SQLMode.HasStrictMode()
+	// those are the default values for lightning CSV format too
+	lineFieldsInfo := plannercore.LineFieldsInfo{
+		FieldsTerminatedBy: `,`,
+		FieldsEnclosedBy:   `"`,
+		FieldsEscapedBy:    `\`,
+		FieldsOptEnclosed:  false,
```

Comment:
this setting is not used. But in case we implement it in future, it should be `true`? which means enclosing by quotes is optional.

Reply:
https://github.com/pingcap/tidb/pull/44204/commits/dc72744fff6c9094383ab71b64ae83252f8f35f7 add a comment